neqo-transport/src/connection.rs

```diff
@@ -784,6 +784,17 @@ impl Connection {
         Ok(())
     }
 
+    fn discarde_keys(&mut self, pn_space: PNSpace) {
+        debug_assert!((pn_space == PNSpace::Initial) || (pn_space == PNSpace::Handshake));
+        let epoch = if pn_space == PNSpace::Initial { 0 } else { 2 };
+        if self.crypto.is_discarded(epoch) {
+            return;
+        }
+        let lost_packets = self.loss_recovery.remove_state_for_pn_space(pn_space);
+        self.handle_lost_packets(&lost_packets);
```

Comment:
When things are dropped, we don't consider the packets to be lost (because that drives congestion windows down).  All we have to do is remove them from bytes in flight.  For that, I think that the loss_recovery code can deal with that.

Reply:
This one is only responsible for marking frames(data in frames) lost and mark them to be retransmitted. This does not change congestion control. While writing this I just realize that I do not need to call this function at all, this data will not be retransmitted.

Looking a bit, I just realize that retry also do not handle bytes in flight properly (my code just use the same function as the retry code and they both are wrong). I will fix both in this PR.